OHHTTPStubs/Sources/OHHTTPStubsResponse.m

```diff
@@ -135,23 +144,47 @@ -(instancetype)initWithFileAtPath:(NSString*)filePath
                        statusCode:(int)statusCode
                           headers:(nullable NSDictionary*)httpHeaders
 {
+    NSURL *fileURL = filePath ? [NSURL fileURLWithPath:filePath] : nil;
+    self = [self initWithFileURL:fileURL
+                      statusCode:statusCode
+                         headers:httpHeaders];
+    return self;
+}
+
+-(instancetype)initWithFileURL:(NSURL *)fileURL
+                    statusCode:(int)statusCode
+                       headers:(nullable NSDictionary *)httpHeaders {
     NSInputStream* inputStream;
-    if (filePath)
+    if (fileURL)
     {
-        inputStream = [NSInputStream inputStreamWithFileAtPath:filePath];
+        // [NSURL -isFileURL] is only available on iOS 8+
+        if (![fileURL.scheme isEqualToString:NSURLFileScheme]) {
+            @throw [NSException exceptionWithName:NSInternalInconsistencyException
+                                           reason:[NSString stringWithFormat:@"Only file URLs may be passed to %@", NSStringFromSelector(_cmd)]
+                                         userInfo:nil];
+        }
+        inputStream = [NSInputStream inputStreamWithURL:fileURL];
     }
     else
     {
-        NSLog(@"%s: nil file path. Returning empty data", __PRETTY_FUNCTION__);
+        NSLog(@"%s: nil URL. Returning empty data", __PRETTY_FUNCTION__);
         inputStream = [NSInputStream inputStreamWithData:[NSData data]];
     }
     
-    NSDictionary* attributes = [NSFileManager.defaultManager attributesOfItemAtPath:filePath error:nil];
-    unsigned long long fileSize = [[attributes valueForKey:NSFileSize] unsignedLongLongValue];
-    self = [self initWithInputStream:inputStream
-                            dataSize:fileSize
-                          statusCode:statusCode
-                             headers:httpHeaders];
+    NSError *error;
+    NSNumber *fileSize;
+    [fileURL getResourceValue:&fileSize forKey:NSURLFileSizeKey error:&error];
+    
+    if (error) {
+        NSLog(@"%s: Error getting file size for URL. Returning empty data", __PRETTY_FUNCTION__);
+        inputStream = [NSInputStream inputStreamWithData:[NSData data]];
+    } else {
+        self = [self initWithInputStream:inputStream
+                                dataSize:[fileSize unsignedLongLongValue]
+                              statusCode:statusCode
+                                 headers:httpHeaders];
+    }
+    
```

Comment:
@AliSoftware From the documentation it sounds like `success` can be `YES`, no error would be set, and the returned value would be nil:

> If this method returns YES and value is populated with nil, it means the resource property is not available for the specified resource and no errors occurred when determining the resource property was not available.

This code really only cares about `fileSize` being non-nil—I think the values of `success` and `error` are only useful for informing the user why the operation failed in this case.